### PR TITLE
fix: environment-scoped uploads

### PR DESCRIPTION
**Pull request by jsdalton**
## Summary

(Re)introduce environment scoped uploads.

## Description

This PR restores environment scope uploads that were introduced here https://github.com/contentful/contentful-management.js/pull/2030 and reverted here https://github.com/contentful/contentful-management.js/pull/2084 

The issue with the original PR was raised here https://github.com/contentful/contentful-management.js/issues/2075 . Further investigation revealed that the problem was in fact an issue with the backend asset processing service. This service was missing a key piece of logic to make it aware that the upload passed for processing could be in a specific environment instead of at the space level. The SDK change revealed that backend issue when it exposed environment scoping to uploads.

## Checklist (check all before merging)

- [ ] Both unit and integration tests are passing
- [ ] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [ ] The new method is exported through the default and plain CMA client
- [ ] All new public types are exported from `./lib/export-types.ts`
- [ ] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

**Review by ryunsong-contentful — APPROVED** [review]
Is there anything with uploadFrom we need to also adjust before merging this? Or is that fixed now with Richard's PR?

**Review by mgoudy91 — APPROVED** [review]

File: lib/adapters/REST/endpoints/asset.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ export const createFromFiles: RestEndpoint<'Asset', 'createFromFiles'> = (
 const ASSET_PROCESSING_CHECK_WAIT = 3000
 const ASSET_PROCESSING_CHECK_RETRIES = 10
 
-function checkIfAssetHasUrl(
+async function checkIfAssetHasUrl(
```
Comment (RichardM99) [inline lib/adapters/REST/endpoints/asset.ts:207]: Why this change (and in some of the other functions) considering there is no `await`?

**Review by RichardM99 — APPROVED** [review]
🥳 

Small note - when I was looking at the customer ticket you mentioned in the [slack thread](https://contentful.slack.com/archives/C059CFR8ZLN/p1702051184713559) I was unable to use the `createFromFiles` method on the SDK without passing the env id, which I think is a bug since it should be possible to use it with space id only.

Not sure if that's due to the changes as part of this PR or not.

**Comment by jsdalton** [comment]
> Small note - when I was looking at the customer ticket you mentioned in the [slack thread](https://contentful.slack.com/archives/C059CFR8ZLN/p1702051184713559) I was unable to use the `createFromFiles` method on the SDK without passing the env id, which I think is a bug since it should be possible to use it with space id only.

@RichardM99  gotcha, but just as a note the type definition requiring the environment id has long existed in this library: https://github.com/contentful/contentful-management.js/blob/master/lib/plain/common-types.ts#L504

so if we want/need to change this, my assumption is we should do it in a separate PR i.e. unrelated to this PR.
